ProcessLib/ThermoMechanics/ThermoMechanicsFEM.h

```diff
@@ -295,6 +295,18 @@ class ThermoMechanicsLocalAssembler
             //
             KuT.noalias() +=
                 B.transpose() * C * alpha * Invariants::identity2 * N * w;
+            if (_process_data.material->getConstitutiveModel() ==
+                MaterialLib::Solids::ConstitutiveModel::CreepBGRa)
```

Comment:
IMHO the addition to KuT can always be assembled. Or a check `if (creep_coefficient != 0` would suffice.
That would remove explicit material model references from the general TM assembly. Then also `getConstitutiveModel()` could be removed again.

Reply:
Because other models may have difference algorithm of that exta  K_uT.

Reply:
I see. Still, I don't really like to mention concrete models explicitly inside a general process. Adding a method `getCouplingDisplacementTemperature(B, C, N, ...)` to the material model is ugly.
Since your solution currently seems to be the most pragmatic one, I'm fine with it.

Reply:
@chleh I agree with you that it is not good to have explicit access to material model inside the assembly, and also to pass B, C, N ... to the material model is ugly. I will consider an encapsulation solution if there are other temperature dependent solid modes are considered on future. At the current stage, I would like to keep this  most pragmatic implementation.